starknet_py/net/full_node_client.py

```diff
@@ -472,3 +462,54 @@ def _get_call_payload(tx: Union[InvokeFunction, Call]) -> dict:
         "entry_point_selector": convert_to_felt(tx.selector),
         "calldata": [convert_to_felt(i) for i in tx.calldata],
     }
+
+
+def _create_broadcasted_txn(transaction: Union[InvokeFunction, Declare]) -> Dict:
+    common_params = {
+        "max_fee": hex(transaction.max_fee),
+        "version": hex(transaction.version),
+        "signature": [convert_to_felt(i) for i in transaction.signature],
+        "nonce": convert_to_felt(transaction.nonce)
+        if transaction.nonce is not None
+        else transaction.nonce,
+    }
+
+    if transaction.tx_type == TransactionType.INVOKE_FUNCTION:
+        invoke_specific_params = {
+            "type": "INVOKE",
+            "calldata": [convert_to_felt(i) for i in transaction.calldata],
+        }
```

Comment:
I get it avoids some duplication but I think it'd be cleared for other developers if we just had
```python
if transaction.version == 0:
  { ... }  # whole v0 transaction
else:
  { ... }  # whole v1 transaction
```
without combining tx version specific and common params at the end.

Reply:
I don't think this is a good idea, because `common_params` are for the `Declare` transaction too

Reply:
Problem is this function is quite complicated to comprehend because parts of transaction come from 3 different places. 

Perhaps there's another way of improving it? Or if you think it's the bast way, we can keep it as is.